iModelCore/ECPresentation/Source/PresentationManagerImpl.cpp

```diff
@@ -1198,7 +1198,9 @@ ContentDescriptorCPtr RulesDrivenECPresentationManagerImpl::_GetNodesDescriptor(
             ),
         *KeySet::Create(params.GetParentNodeKey() ? NavNodeKeyList{ params.GetParentNodeKey() } : NavNodeKeyList{})
         ), params);
-    return GetContentDescriptor(descriptorParams);
+    auto descriptor = GetContentDescriptor(descriptorParams);
+    (const_cast<ContentDescriptor*>(descriptor.get()))->SetHierarchyLevelRuleset(*ruleset);

```

Comment:
This is not good. The descriptor we get from `GetContentDescriptor` possibly comes from cache, so we definitely don't want to mutate it. Instead, we could clone it using `ContentDescriptor::Create(*sourceDescriptor)` and then modify.

Reply:
fixed